src/theory/quantifiers/candidate_rewrite_filter.cpp

```diff
@@ -379,18 +384,29 @@ bool CandidateRewriteFilter::notify(Node s,
     for (unsigned i = 0, size = vars.size(); i < size; i++)
     {
       Trace("crf-match") << "    " << vars[i] << " -> " << subs[i] << std::endl;
-      // TODO (#1923) ensure that we use an internal representation to
-      // ensure polymorphism is handled correctly
-      Assert(vars[i].getType().isComparableTo(subs[i].getType()));
     }
   }
+#ifdef CVC4_ASSERTIONS
```

Comment:
Note to self: we should have a way to check this without resorting to `#ifdef`s.